### PR TITLE
fix: tokenData are passed now from AddTokenByNetwork consistent with the chosen network

### PR DESCRIPTION
**Pull request by AntonioVentilii-DFINITY**
# Motivation

To be more consistent with the selected network from the dropdown, the data returned are set when the component is destroyed.

# Changes

Prop `tokenData` is defined with `onDestroy`.

### CODE REVIEW

File: src/frontend/src/icp-eth/components/tokens/AddTokenByNetwork.svelte [hunk 1]
```diff
@@ -23,11 +24,13 @@
 	let indexCanisterId: string;
 	let erc20ContractAddress: string;
 
-	$: tokenData = {
-		ledgerCanisterId,
-		indexCanisterId,
-		erc20ContractAddress
-	};
+	onDestroy(() => {
+		tokenData = isNetworkIdICP(network?.id)
```
Comment (peterpeterparker) [inline src/frontend/src/icp-eth/components/tokens/AddTokenByNetwork.svelte:28]: Can you add a comment about why we do this as it's quite an original pattern?

**Review by peterpeterparker — APPROVED** [review]
LGTM, thx

**Comment by AntonioVentilii-DFINITY** [comment]
@peterpeterparker check the new version, maybe it is better?

p.s. I tried to do inline, 

```typescript
$: tokenData = isNetworkIdICP(network?.id)
		? { ledgerCanisterId, indexCanisterId }
		: isNetworkIdEthereum(network?.id)
			? { erc20ContractAddress }
			: {};		
```
but it was giving an error because it was interpreting this as assigning undefined keys

File: src/frontend/src/icp-eth/components/tokens/AddTokenByNetwork.svelte (hunk 1 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/icp-eth/components/tokens/AddTokenByNetwork.svelte]: check now, I updated the logic, maybe it is better?

**Review by peterpeterparker — APPROVED** [review]
I like it better!

Thanks

**Pull request merged**